branch-3.0: [fix](nereids) fix distinct window compute wrong result (#48987)

924060929 opened the pull request:
cherry pick from #48987

924060929 committed:
cherry pick from #48987

fix distinct window compute wrong result, introduced by #14397
```sql
select distinct sum(value) over(partition by id)
from (
  select 100 value, 1 id
  union all
  select 100, 2
)a;

+----------------------------------+
| sum(value) over(partition by id) |
+----------------------------------+
| 100                              |
| 100                              |
+----------------------------------+
```

hello-stephen commented:
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

924060929 commented:
run buildall

924060929 committed:
cherry pick from #48987

fix distinct window compute wrong result, introduced by #14397
```sql
select distinct sum(value) over(partition by id)
from (
  select 100 value, 1 id
  union all
  select 100, 2
)a;

+----------------------------------+
| sum(value) over(partition by id) |
+----------------------------------+
| 100                              |
| 100                              |
+----------------------------------+
```

924060929 committed:
cherry pick from #48987

fix distinct window compute wrong result, introduced by #14397
```sql
select distinct sum(value) over(partition by id)
from (
  select 100 value, 1 id
  union all
  select 100, 2
)a;

+----------------------------------+
| sum(value) over(partition by id) |
+----------------------------------+
| 100                              |
| 100                              |
+----------------------------------+
```

924060929 commented:
run buildall

924060929 committed:
cherry pick from #48987

fix distinct window compute wrong result, introduced by #14397
```sql
select distinct sum(value) over(partition by id)
from (
  select 100 value, 1 id
  union all
  select 100, 2
)a;

+----------------------------------+
| sum(value) over(partition by id) |
+----------------------------------+
| 100                              |
| 100                              |
+----------------------------------+
```

924060929 commented:
run buildall

924060929 committed:
opt

924060929 commented:
run buildall
